src/novo/src/client.tsx

```diff
@@ -2,27 +2,23 @@ import "desktop/lib/webpackPublicPath"
 
 import React from "react"
 import ReactDOM from "react-dom"
-import { buildClientApp } from "v2/Artsy/Router/client"
 import { getAppNovoRoutes } from "v2/Apps/getAppNovoRoutes"
 import { loadableReady } from "@loadable/component"
-import { initModalManager } from "desktop/apps/authentication/client/initModalManager"
 import { logoutEventHandler } from "desktop/lib/logoutHandler"
 import { mediator } from "lib/mediator"
 
-buildClientApp({
-  routes: getAppNovoRoutes(),
-})
+import("v2/Artsy/Router/client")
+  .then(({ buildClientApp }) => {
+    return buildClientApp({
+      routes: getAppNovoRoutes(),
+    })
+  })
   .then(({ ClientApp }) => {
     loadableReady(() => {
       ReactDOM.hydrate(<ClientApp />, document.getElementById("react-root"))
     })
   })
-  .then(() => {
-    initModalManager()
-  })
-  .then(() => {
-    mediator.on("auth:logout", logoutEventHandler)
-  })
-  .catch(error => {
-    console.error(error)
-  })
+  .then(() => { return import("desktop/apps/authentication/client/initModalManager") })
```

Comment:
this is kinda strange too

Reply:
One extra step, I'm not sure we need the loadable wrapper here and loadable uses import for chunking anyway. If we needed to use some extra features such as loadables manual preload I could see a case for it.

This is also only being done to test if splitting the large blocking js task of 600ms into two makes a difference. If so there will be a lot more chunking to come and this will all look very different.

Reply:
Makes sense 👍